pallets/liquid-staking-two/src/lib.rs

```diff
@@ -0,0 +1,820 @@
+                            status: ResponseStatus::Successed,
+                            ..op
+                        })
+                        .ok_or(Error::<T>::OperationNotReady)?;
+                    *op = Some(next_op.clone());
+                    Ok(next_op)
+                },
+            )?;
+            T::Currency::deposit(T::LiquidCurrency::get(), &Self::account_id(), op.amount)?;
+            Self::deposit_event(Event::<T>::BondSucceed(era_index));
+            Ok(().into())
+        }
+
+        /// Invoked when unbonding extrinsic finished. Burn previously transfered xksm if
+        /// successed.
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn record_unbond_response(
+            origin: OriginFor<T>,
+            era_index: Option<EraIndex>,
+        ) -> DispatchResultWithPostInfo {
+            let _who = ensure_signed(origin)?;
+            let era_index = era_index.unwrap_or(Self::current_era());
+
+            let op = StakingOperationHistory::<T>::try_mutate(
+                &era_index,
+                StakingOperationType::Unbond,
+                |op| -> Result<Operation<_>, DispatchError> {
+                    let next_op = op
+                        .clone()
+                        .filter(|op| op.status == ResponseStatus::Ready)
+                        .map(|op| Operation {
+                            status: ResponseStatus::Successed,
+                            ..op
+                        })
+                        .ok_or(Error::<T>::OperationNotReady)?;
+                    *op = Some(next_op.clone());
+                    Ok(next_op)
+                },
+            )?;
+            T::Currency::withdraw(T::LiquidCurrency::get(), &Self::account_id(), op.amount)?;
+            Self::deposit_event(Event::<T>::UnbondSucceed(era_index));
+            Ok(().into())
+        }
+
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn transfer_to_relaychain(
+            origin: OriginFor<T>,
+            amount: Balance,
+            agent: T::AccountId,
+        ) -> DispatchResultWithPostInfo {
+            let _who = ensure_signed(origin)?;
+            // TODO(Alan WANG): Check agent is approved.
+            T::XcmTransfer::transfer(
+                Self::account_id(),
+                T::StakingCurrency::get(),
+                amount,
+                MultiLocation::X2(
+                    Junction::Parent,
+                    Junction::AccountId32 {
+                        network: NetworkId::Any,
+                        id: agent.into(),
+                    },
+                ),
+                T::BaseXcmWeight::get(),
+            )?;
+            Ok(().into())
+        }
+
+        // this method must be called after update the era index, so we use previous era to calculate
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn deposit_event_to_relaychain(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
```

Comment:
shall this function be called before next era and use current era? so that users can start to earn rewards in next era

Reply:
it is used for calculate the previous era.